ntoskrnl/cc/view.c

```diff
@@ -1142,6 +1142,8 @@ CcFlushCache (
         IoStatus->Information = 0;
     }
 
+    KeAcquireGuardedMutex(&SharedCacheMap->FlushCacheLock);
```

Comment:
Is this new lock meant to be so that there are no two or more concurrent flushes for perhaps intersecting memory regions (via their VACBs) happening at the same time?

Reply:
Is that race-condition also possible on FAT? If not, why?

Reply:
It's not happen with FAT32 partition using the files copy test (in the video). The results from the debugger indicates that fastfat rarely calls CcFlushCache but BTRFS always call it in the cleanup routine.

Reply:
yes, my question targetted whether it was possible to happen on FAT **before** the commit that I linked.

Reply:
> Without this lock, these concurrent flushes will result in inconsistency of the VACB struct 

The VACB structs aren't locked already via the `CcRosLookupVacb` call?

Reply:
> whether it was possible to happen on FAT before the commit that I linked.

Not tested yet. I think the bug [CORE-17624](https://jira.reactos.org/browse/CORE-17624) is unrelated to this bug because most of the Cc's VACB management functions have asserts. When this bug happens, it always trigger assertion failures.
This patch has no visible effect with the FAT32 partition.

Reply:
Thx for expressing your thoughts. My questions in here have been answered.